Code/Framework/AzToolsFramework/Platform/Common/UnixLike/AzToolsFramework/API/PythonLoader_UnixLike.cpp

```diff
@@ -7,16 +7,17 @@
  */
 
 #include <AzToolsFramework/API/PythonLoader.h>
+#include <AzToolsFramework_Traits_Platform.h>
 #include <AzCore/Debug/Trace.h>
 #include <AzCore/IO/Path/Path.h>
+
 #include <dlfcn.h>
 
 namespace AzToolsFramework::EmbeddedPython
 {
-
-
-    PythonLoader::PythonLoader()
+    void PythonLoader::LoadRequiredModules()
     {
+        #if AZ_TRAIT_PYTHON_LOADER_ENABLE_EXPLICIT_LOADING
```

Comment:
It seem like the PythonLoader class can use the [DynamicModuleHandle](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Module/DynamicModuleHandle.h#L33-L35) class to store an RAII handle to the loaded Python dll as a member variable.

In the PythonLoader.h file an `AZStd::unique_ptr<DynamicModuleHandle> m_embeddedLibPythonHandle` can be stored.

Then inside of this function a call to [DynamicModuleHandle::Create](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Module/DynamicModuleHandle.h#L33-L35) can be called followed by `DynamicModuleHandle::Load`.

Finally in `UnloadRequiredModule`  a call to `DynamicModuleHandle::Unload` can be invoked

Reply:
This was done because Linux implementation of [AZ::Platform::OpenModule ](https://github.com/spham-amzn/o3de/blob/d6f0da68cb6585591959771cef8a896e9a6fd13d/Code/Framework/AzCore/Platform/Linux/AzCore/Module/DynamicModuleHandle_Linux.cpp#L20-L29) will only pass in the `RTLD_NOW` flag. 

This specific [module ](https://github.com/spham-amzn/o3de/blob/d5fd83edcd13259ea87ab4ffd4b95a26b5e1716f/Code/Framework/AzToolsFramework/Platform/Common/UnixLike/AzToolsFramework/API/PythonLoader_UnixLike.cpp#L26) loading needed to use the RTLD_GLOBAL flag which is not done for the standard O3DE module loading, otherwise the python binding modules wouldn't be able to resolve the symbols.